raven-android/src/main/java/com/getsentry/raven/android/Raven.java

```diff
@@ -0,0 +1,178 @@
+package com.getsentry.raven.android;
+
+import android.Manifest;
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.text.TextUtils;
+import android.util.Log;
+import com.getsentry.raven.DefaultRavenFactory;
+import com.getsentry.raven.RavenFactory;
+import com.getsentry.raven.dsn.Dsn;
+import com.getsentry.raven.event.Event;
+import com.getsentry.raven.event.EventBuilder;
+
+/**
+ * Android specific class to interface with Raven. Supplements the default Java classes
+ * with Android specific state and features.
+ */
+public final class Raven {
+
+    /**
+     * Logger tag.
+     */
+    public static final String TAG = Raven.class.getName();
+
+    private static volatile com.getsentry.raven.Raven raven;
+
+    /**
+     * Hide constructor.
+     */
+    private Raven() {
+
+    }
+
+    /**
+     * Initialize Raven using a DSN set in the AndroidManifest.
+     *
+     * @param ctx Android application ctx
+     */
+    public static void init(Context ctx) {
+        ctx = ctx.getApplicationContext();
+        String dsn = "";
+
+        // attempt to get DSN from AndroidManifest
+        ApplicationInfo appInfo = null;
+        try {
+            PackageManager packageManager = ctx.getPackageManager();
+            appInfo = packageManager.getApplicationInfo(ctx.getPackageName(), PackageManager.GET_META_DATA);
+            dsn = appInfo.metaData.getString("com.getsentry.raven.android.DSN");
+        } catch (PackageManager.NameNotFoundException e) {
+            // skip
+        }
+
+        if (TextUtils.isEmpty(dsn)) {
+            throw new NullPointerException("Raven DSN is not set, you must provide it via"
+                + "the constructor or AndroidManifest.");
+        }
+
+        init(ctx, new Dsn(dsn));
+    }
+
+    /**
+     * Initialize Raven using a string DSN.
+     *
+     * @param ctx Android application ctx
+     * @param dsn Sentry DSN string
+     */
+    public static void init(Context ctx, String dsn) {
+        init(ctx, new Dsn(dsn));
+    }
+
+    /**
+     * Initialize Raven using a DSN object. This is the 'main' initializer that other methods
+     * eventually call.
+     *
+     * @param ctx Android application ctx
+     * @param dsn Sentry DSN object
+     */
+    public static void init(Context ctx, Dsn dsn) {
+        if (raven != null) {
+            Log.e(TAG, "Initializing Raven multiple times.");
+            // cleanup existing connections
+            raven.closeConnection();
```

Comment:
I used to throw here. The problem is that Android Studio "run" will do hot code replacement and I've found that means `Raven.init` will just be recalled on my existing process, causing the app to crash. Since `Raven.init` will be common in startup code and Android Studio is the official IDE, I thought this was crap UX.

So, since all Raven access on Android is static, I cleanup the old connection before I `re-init`. The reason I don't just `return` is because people could be changing their `Dsn` settings and hitting "run"...